authors/apps/articles/models.py

```diff
@@ -98,95 +78,42 @@ def save(self, *args, **kwargs):
-        self.name = username
-        self.article_id = self.article.id
-
-    def like_article(self, username, slug, *args, **kwargs):
-
-        self.get_article(username, slug)
-
-        #Check if the user has already liked the article
-        try:
-            a_like = Likes.objects.get(
-                name=self.name, article_id=self.article_id, like_dislike='L'
-            )
-            if isinstance(a_like, object):
-                message = 'You have already liked this article.'
-                raise exceptions.PermissionDenied(message)
-        except Likes.DoesNotExist:
-            updated_likes = self.article.likes + 1
-            Article.objects.filter(slug=slug).update(
-                likes=updated_likes
-            )
-            self.like_dislike = 'L'
-            return self.save()
-
-    @staticmethod
-    def unlike(username, slug):
-
-        try:
-            article = Article.objects.get(slug=slug)
-            an_unlike = Likes.objects.get(
-                name=username, article_id=article.id, like_dislike='L'
-            )
-            updated_likes = article.likes - 1
-            Article.objects.filter(slug=slug).update(
-                likes=updated_likes
-            )
-            return an_unlike.delete()
-        except Likes.DoesNotExist:
-            message = 'You have not yet liked this article.'
-            raise exceptions.PermissionDenied(message)
-
-    def dislike(self, username, slug):
-
-        self.get_article(username, slug)
-        try:
-            a_dislike = Likes.objects.get(
-                name=self.name, article_id=self.article_id, like_dislike='D'
-            )
-            if isinstance(a_dislike, object):
-                message = 'You have already disliked this article.'
-                raise exceptions.PermissionDenied(message)
-        except Likes.DoesNotExist:
-            updated_dislikes = self.article.dislikes + 1
-            Article.objects.filter(slug=slug).update(
-                dislikes=updated_dislikes
-            )
-            self.like_dislike = 'D'
-            return self.save()
-
-    @staticmethod
-    def un_dislike(username, slug):
-
-        try:
-            article = Article.objects.get(slug=slug)
-            un_dislike = Likes.objects.get(
-                name=username, article_id=article.id, like_dislike='D'
-            )
-            updated_dislikes = article.dislikes - 1
-            Article.objects.filter(slug=slug).update(
-                dislikes=updated_dislikes
-            )
-            return un_dislike.delete()
-        except Likes.DoesNotExist:
-            message = 'You have not yet disliked this article.'
-            raise exceptions.PermissionDenied(message)
-        # import pdb;pdb.set_trace()
+class Impression(models.Model):
+    name = models.CharField(max_length=50)
+    description = models.CharField(max_length=100)
+    image = models.URLField(blank=True)
+
+    def like_impression(self):
```

Comment:
Please remove the `_impression` from these three methods. It does not add anything context-wise. These methods are in the Impression model so whoever is reading this code can quickly tell that they are connected to the Impression "thing".